### PR TITLE
fix(neorg): resolve workspace notation with image files

### PR DESCRIPTION
**Pull request by pysan3**
>> Is this allowed? (image.nvim does not support this atm)
>> ```txt
>> .image $/relative/path.png (the `$` reference to workspace)
>> ```
>
> it is allowed, yep! obviously image.nvim will have some difficulties with it because it's not interacting with neorg so it can't possible know the workspace root
>
> --Copied from discord.

This PR adds support for `.image $<workspace>/path/to/image.png`.

Obviously, we don't want neorg as a hard dependency of this plugin so I tried my best to not cause any delay on users without `core.dirman` module imported, and encapsulated `require("neorg.core")` inside a `pcall`.

This PR is currently in draft as it needs https://github.com/nvim-neorg/neorg/pull/1143 to be merged beforehand. I'll ping you again when the above PR is merged.

- https://github.com/nvim-neorg/neorg/pull/1143

### CODE REVIEW

**Comment by 3rd** [comment]
Hey, thanks for the PR, looking good!

**Comment by pysan3** [comment]
Hi @3rd ,

The related PR in neorg has been merged to main branch and is ready to go.

It will be included in the v6.1.1 release.
https://github.com/nvim-neorg/neorg/pull/1140

File: lua/image/integrations/neorg.lua [hunk 1]
```diff
@@ -32,6 +49,15 @@ return document.create_document_integration({
           range = { start_row = start_row, start_col = start_col, end_row = end_row, end_col = end_col },
           url = path,
         })
+
+        local ok, workspace_path = maybe_parse_workspace_path(path)
+        if ok and workspace_path ~= path then
+          table.insert(images, {
```
[the remark this answers is not in the document]
Comment (pysan3) [reply lua/image/integrations/neorg.lua:55]: Unfortunately, the neorg return value is not 100% correct on Windows user (image.nvim might not work on windows in the first place but anyways). 

So I don't want always to overwrite the given path. 

If I understand correctly, the paths added to the table are checked it's existence afterwards, so I decided to just inject the result from neorg. If it is the workspace notation, raw path will never exist, and if it is a legitimate path, result from neorg will be wrong and will be always ignored. 

Sometimes neorg returns the same value as given, but that's the case when I didn't append the same path in the table with `path ~= workspace_path`

[the remark this answers is not in the document]
Comment (3rd) [reply lua/image/integrations/neorg.lua:55]: Yep, there's no support for Windows, and there will never be :smiling_imp: 
I guess you're right, no harm in the current implementation, merging.
Thanks for all the effort!

**Pull request merged**